app/presenters/coronavirus/sub_section_json_presenter.rb

```diff
@@ -29,70 +14,27 @@ def output
     @output ||=
       {
         title: title,
-        sub_sections: sub_sections,
+        sub_sections: create_sub_sections,
       }
   end
 
-  def sub_sections
-    content_groups.map { |content_group| sub_section_hash_from_content_group(content_group) }
-  end
-
-  # Groups the sub section content into an array of arrays, such that:
-  #   - links and texts are separated and each is put into an inner array
-  #   - each title starts a new inner array containing that title as the first element
-  #     and then the links between that title and the next title (or the end of the content)
-  #   - if there are no titles, all the links go into a single inner array
-  def content_groups
-    sub_section.content.lines.each_with_object([]) do |line, sections|
-      sections << [] if sections.empty? || is_header?(line)
-      line.strip!
-      sections.last << line
-    end
-  end
-
-  # Converts:
-  #
-  #   [header, link, link]
-  #
-  # into:
-  #
-  #    {
-  #      title: <text from header>,
-  #      list: [
-  #        {
-  #          label: <first link label>,
-  #          url: <first link url>
-  #        },
-  #        {
-  #          label: <second link label>,
-  #          url: <second link url>
-  #        }
-  #    {
-  def sub_section_hash_from_content_group(content_group)
-    content_group.each_with_object({ title: nil }) do |line, hash|
-      if is_header?(line)
-        title = HEADER_PATTERN.match(line).named_captures["title"]
-        hash[:title] = title
-      elsif is_link?(line)
-        hash[:list] ||= []
-        hash[:list] << build_link(line)
-      else
-        raise MarkdownInvalidError, "Unable to parse markdown: '#{line}'"
+  def create_sub_sections
+    sub_section.structured_content.items.each_with_object([]) do |item, memo|
+      case item
+      when Coronavirus::SubSection::StructuredContent::Link
+        if memo.empty?
+          memo << { title: nil, list: [build_link(item.label, item.url)] }
+        else
+          memo.last[:list] << build_link(item.label, item.url)
+        end
+      when Coronavirus::SubSection::StructuredContent::Header
+        memo << { title: item.text, list: [] }
```

Comment:
This is much nicer!

Reply:
Thank you :) just going to fix up the commit messages now!